staging/src/k8s.io/component-base/logs/registry.go

```diff
@@ -32,21 +32,30 @@ var logRegistry = NewLogFormatRegistry()
 // LogFormatRegistry store klog format registry
 type LogFormatRegistry struct {
 	registry map[string]logr.Logger
+	freezed  bool
 }
 
 // NewLogFormatRegistry return new init LogFormatRegistry struct
 func NewLogFormatRegistry() *LogFormatRegistry {
 	return &LogFormatRegistry{
 		registry: make(map[string]logr.Logger),
+		freezed:  false,
 	}
 }
 
 // Register new log format registry to global logRegistry
 func (lfr *LogFormatRegistry) Register(name string, logger logr.Logger) error {
+	if lfr.freezed {
```

Comment:
I think we should protect all mutations, please add this to Set and Delete

Reply:
@serathius , got it.
Btw, any changes needed in error string?